### ISSUE TITLE
Triangular PDF Noise (as alternative to photon noise)

### ISSUE THREAD

**Issue by Matt-Gore** [libjxl/libjxl#900]
Photon noise is nice but it doesn't fit everywhere. For example, digital artworks might use more "consistent" noise with regards to brightness. If you try to compensate for the loss (due to compression) of such noise by adding back photon_noise, it will change the noise characteristics and deviate (unnecessarily) from the original.

It would thus be nice if we could have a triangular PDF noise as alternative for these use-cases, especially since it should be easy to implement such a simple PDF.

**Comment by jonsneyers** [comment]
Question is not how easy it is to implement, but if it can be represented in the jxl bitstream or not. Any thoughts on that, @sboukortt ?

**Comment by sboukortt** [comment]
To me, there seem to be two orthogonal questions here: the distribution (triangular or not), and the function that maps image brightness to noise amplitude. The former, we cannot change (but from what I remember, it’s already somewhat close to triangular), while the latter is an 8-entry LUT.  `--photon_noise` creates a LUT with a certain “shape” but indeed there are other possibilities. It should be possible to write a LUT that represents a flatter brightness->noise amplitude mapping if that is desired.

Do I understand correctly that this is more about that mapping?

**Comment by Matt-Gore** [comment]
Yes, if the noise is triangular-like then that's already good. So the important part would be that the noise amplitude is consistent across the sRGB colorspace/brightness.
